src/components/GoalPage/GoalPage.tsx

```diff
@@ -191,6 +193,32 @@ export const GoalPage = ({ user, ssrTime, params: { id } }: ExternalPageProps<{
         [goal, invalidateFn, addParticipantMutation],
     );
 
+    const onAssigneeChange = useCallback(
+        async (activity?: ActivityByIdReturnType) => {
+            if (goal && activity && activity.user) {
+                setCurrentOwner(activity?.user || currentOwner);
+                await update({
```

Comment:
It must be try/catch to resolve unexpected result via `setCurrentOwner(currentOwner)`